msteams-platform/toolkit/debug-your-Teams-app-test-tool.md

```diff
@@ -0,0 +1,289 @@
+
+1. **Sandbox environment**: Sandbox environment emulates the behavior, look and user experience of Teams.
+
+1. **Tunneling**: Tunnel service is not required as it runs locally.
+
+1. **Reduce account dependencies**: Microsoft 365 Developer tenant and the app uploading permissions aren't required to debug the bot application.
+
+1. **Rapid inner-loop iterations**: Optimizes the process of making changes to the app design and bot logic without having to redeploy the bot application to the cloud.
+
+1. **Mock data and activities**: The Test Tool makes it easy to accomplish a test for complex scenarios such as **Send a welcome message when a new member joins the channel** by using mock data and built-in and custom activity triggers.
+
+1. **Reliable and trustable**: The Test Tool is reliable as the Bot application [Adaptive Card](../task-modules-and-cards/cards/cards-reference.md#adaptive-card) utilizes the same rendering technology as in Teams.
+
+1. **Integration with existing Teams bot applications**: The Test Tool can be integrated with existing Teams bot applications that are built with [Bot Framework SDK](https://dev.botframework.com/).
+
+1. **Support for different scopes**: The Test Tool supports testing in personal, team, and group chat scopes.
+
+### Limitations
+
+The following are the limitations:
+
+1. The Test Tool doesn't process the app manifest, which means features that are only enabled through the manifest won't be accessible.
+
+1. The Test Tool doesn't support all other types of Bot application [Cards](../task-modules-and-cards/what-are-cards.md#cards).
+
+1. The Test Tool doesn't support the following Adaptive Card features:
+
+   * [People Picker](../task-modules-and-cards/cards/people-picker.md)
+   * [Typeahead search](../task-modules-and-cards/cards/dynamic-search.md)
+   * [User mention](../task-modules-and-cards/cards/cards-format.md#microsoft-azure-active-directory-azure-ad-object-id-and-upn-in-user-mention)
+   * [Stage View](../task-modules-and-cards/cards/cards-format.md#stage-view-for-images-in-adaptive-cards)
+   * [Full width](../task-modules-and-cards/cards/cards-format.md#full-width-adaptive-card)
+
+1. The Test Tool doesn't support the following:
+   * Single sign-on (SSO)
+   * Mobile
+   * Meetings
+
+1. **Limited emulation**
+
+   | Features | Debug in Test Tool | [Debug your Teams app locally](debug-local.md) |
+   | --- | --- | --- |
+   | Basic sending / receiving messages | Available | Available |
+   | Bot Framework APIs (TeamsInfo.getPagedMembers()...) |Available (respond with mocked data) | Available |
+   | Sending Teams events | Available (mocking activity) |Available |
+   | Typing indicator | Not Available | Available |
+   | Tab, Messaging Extension, Task Module, SSO, and non-Adaptive Cards | Not Available | Available |
+
+### Teams Toolkit samples and templates
+
+Test the bot sample, which displays the features supported by the Teams App Test Tool, as follows:
```

Comment:
it doesn't work if follow the steps, could you help to refine it?

Reply:
For reference:

1. Open [TeamsFx-Samples repo](https://github.com/OfficeDev/TeamsFx-Samples/tree/v3).
1. Clone the repo locally.
1. Open the sub-folder `test-tool-sample-app` in Visual Studio Code with Teams Toolkit.
1. ...

Reply:
Updated.